addons/attach/functions/fnc_refreshItem.sqf

```diff
@@ -0,0 +1,70 @@
+/*
+ * Author: Phyma
+ * Remove a item of the unit when it is about to die
+ *
+ * Arguments:
+ * 0: vehicle that it will be attached to (player or vehicle) <OBJECT>
+ * 1: unit doing the attach (player) <OBJECT>
+ * 2: classname of the item you want to remove <STRING>
+ * 3: Position to be placed <ARRAY>
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Example:
+ * [bob, bob, "light", [0, 0, 0]] call ace_attach_fnc_removeItemEvent;
+ *
+ * Public: No
+ */
+
+#include "script_component.hpp"
+
+#define LIFETIMEOFFSET 5
+
+params ["_attachToVehicle", "_unit", "_itemClassname", "_position"];
+
+private _itemVehClass = getText (configFile >> "CfgMagazines" >> _itemClassname >> "ACE_Attachable");
+private _timeLive = getNumber (configfile >> "CfgAmmo" >> _itemVehClass >> "timeToLive");
+
+if (0 < _timeLive && _timeLive < 1500) then {
+    //Make sure we remove  it before it is consumed
+    _timeLive = _timeLive - LIFETIMEOFFSET;
+
+    [{
+        params ["_args", "_handle"];
+        _args params ["_attachToVehicle", "_unit", "_itemClassname", "_position"];
+        private ["_attachedObject"];
```

Comment:
This variable is not needed outside of the forEach loop.
Also `private _attachedObject = objNull;` is probably faster.